webviz_subsurface/plugins/_pvt_plot.py

```diff
@@ -0,0 +1,640 @@
+* One column named `VOLUMEFACTOR` as the first covariate.
+* One column named `VISCOSITY` as the second covariate.
+
+The file can e.g. be dumped to disc per realization by a forward model in ERT using
+`ecl2data_frame` or `pyscal`.
+"""
+
+    PHASES = {"OIL": "PVTO", "GAS": "PVDG", "WATER": "PVTW"}
+
+    def __init__(
+        self,
+        app,
+        ensembles: List[str],
+        pvt_relative_file_path: str = None,
+        read_from_init_file: bool = False,
+    ):
+
+        super().__init__()
+
+        self.ensemble_paths = {
+            ensemble: app.webviz_settings["shared_settings"]["scratch_ensembles"][
+                ensemble
+            ]
+            for ensemble in ensembles
+        }
+
+        self.plotly_theme = app.webviz_settings["theme"].plotly_theme
+
+        self.pvt_relative_file_path = pvt_relative_file_path
+
+        if self.pvt_relative_file_path is not None:
+            # Load data from all ensembles into a Panda DataFrame
+            self.pvt_data_frame = load_csv(
+                ensemble_paths=self.ensemble_paths, csv_file=self.pvt_relative_file_path
+            )
+
+            # Rename "type" header column to "keyword" and make all headers having capital letters
+            self.pvt_data_frame = self.pvt_data_frame.rename(
+                str.upper, axis="columns"
+            ).rename(columns={"TYPE": "KEYWORD", "RS": "GOR"})
+
+            # Ensure that the identifier string "KEYWORD" is contained in the header columns
+            if "KEYWORD" not in self.pvt_data_frame.columns:
+                raise ValueError(
+                    "There has to be a KEYWORD or TYPE column with corresponding Eclipse keyword."
+                )
+
+            valid_columns = [
+                "ENSEMBLE",
+                "REAL",
+                "KEYWORD",
+                "PVTNUM",
+                "GOR",
+                "PRESSURE",
+                "VOLUMEFACTOR",
+                "VISCOSITY",
+            ]
+
+            self.pvt_data_frame = self.pvt_data_frame[
+                [
+                    column
+                    for column in self.pvt_data_frame.columns
+                    if column in valid_columns
+                ]
+            ]
+        else:
+            self.pvt_data_frame = load_pvt_dataframe(
+                self.ensemble_paths, use_init_file=read_from_init_file
+            )
+            self.pvt_data_frame = self.pvt_data_frame.rename(
+                str.upper, axis="columns"
+            ).rename(columns={"TYPE": "KEYWORD", "RS": "GOR"})
+
+        self.set_callbacks(app)
+
+    @property
+    def phases(self) -> List[str]:
+        return PvtPlot.PHASES.keys()
+
+    @property
+    def ensembles(self) -> List[str]:
+        return list(self.pvt_data_frame["ENSEMBLE"].unique())
+
+    @property
+    def pvtnums(self) -> List[str]:
+        return list(self.pvt_data_frame["PVTNUM"].unique())
+
+    @property
+    def ensemble_colors(self) -> List[str]:
```

Comment:
I think it is something like this?
```suggestion
    def ensemble_colors(self) -> Dict[str, List[str]]:
```

Reply:
True.